unikernel.ml

```diff
@@ -490,39 +497,186 @@ struct
+                                  "{\"status\": 200, \"success\": true, \
+                                   \"message\": {\"user\": "
+                                  ^ Yojson.Basic.to_string
+                                      (User_model.user_to_json user)
+                                  ^ "}}"
+                                in
+                                let cookie =
+                                  List.find
+                                    (fun (c : User_model.cookie) ->
+                                      c.name = "molly_session")
+                                    user.cookies
+                                in
+                                let cookie_value =
+                                  cookie.name ^ "=" ^ cookie.value
+                                  ^ ";Path=/;HttpOnly=true"
+                                in
+                                let header_list =
+                                  [
+                                    ("Set-Cookie", cookie_value);
+                                    ("location", "/dashboard");
+                                  ]
+                                in
+                                Lwt.return
+                                  (reply ~header_list
+                                     ~content_type:"application/json" res)
+                            | Error (`Msg _msg) ->
+                                let res =
+                                  "{\"status\": 400, \"success\": false, \
+                                   \"message\": \"Something went wrong. Wait a \
+                                   few seconds and try again.\"}"
+                                in
+                                Lwt.return
+                                  (reply ~content_type:"application/json" res)))
+                    ))
+            | _ ->
+                let res =
+                  "{\"status\": 400, \"success\": false, \"message\": \"Bad \
+                   request method\"}"
+                in
+                Lwt.return (reply ~content_type:"application/json" res))
+        | "/api/login" -> (
+            let request = Httpaf.Reqd.request reqd in
+            match request.meth with
+            | `POST -> (
+                decode_request_body reqd >>= fun data ->
+                let json =
+                  try Ok (Yojson.Basic.from_string data)
+                  with Yojson.Json_error s -> Error (`Msg s)
+                in
+                match json with
+                | Error (`Msg s) ->
+                    Logs.warn (fun m -> m "Failed to parse JSON: %s" s);
+                    let res =
+                      "{\"status\": 400, \"message\": \"Bad request body\"}"
+                    in
+                    Lwt.return (reply ~content_type:"application/json" res)
+                | Ok json -> (
+                    let validate_email_re =
+                      Re.Perl.re "[a-zA-Z0-9.$_!]+@[a-zA-Z0-9]+\\.[a-z]{2,3}"
```

Comment:
let's add a TODO here to validate the mail address (with @dinosaure libraries, and maybe later also to send an email with a unique link). for this PR, the code is good :)